engine/src/main/java/org/terasology/rendering/opengl/FrameBuffer.java

```diff
@@ -0,0 +1,428 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.opengl;
+
+import gnu.trove.map.TObjectIntMap;
+import gnu.trove.map.hash.TObjectIntHashMap;
+
+import java.util.Map;
+
+import static org.lwjgl.opengl.GL11.*;
+import static org.lwjgl.opengl.GL11.GL_FLOAT;
+import static org.lwjgl.opengl.GL11.GL_INT;
+import static org.lwjgl.opengl.GL12.*;
+import static org.lwjgl.opengl.GL12.GL_UNSIGNED_INT_10_10_10_2;
+import static org.lwjgl.opengl.GL12.GL_UNSIGNED_INT_2_10_10_10_REV;
+import static org.lwjgl.opengl.GL13.*;
+import static org.lwjgl.opengl.GL21.GL_SRGB8;
+import static org.lwjgl.opengl.GL21.GL_SRGB8_ALPHA8;
+import static org.lwjgl.opengl.GL30.*;
```

Comment:
@pollend: Terasology's current system requirements include support for OpenGL 2.1 and a few additional extensions. You can find the checks in LwjglGraphics.checkOpenGL(). There is certainly the will to eventually support more modern OpenGL releases, especially via alternative renderers. But for the time being OpenGL 2.1 it's all we can use.